packages/plugins/graphql-sse/src/index.ts

```diff
@@ -23,60 +21,116 @@ export interface GraphQLSSEPluginOptions
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const ctxForReq = new WeakMap<Request, any>()
-  let handler!: (request: Request) => Promise<Response>
+  const { pubsub } = options
+  const tokenByRequest = new WeakMap<Request, string>()
+  const operationIdByRequest = new WeakMap<Request, string>()
   return {
-    onYogaInit({ yoga }) {
-      handler = createHandler(
-        {
-          ...handlerOptions,
-          async onSubscribe(req, params) {
-            const enveloped = yoga.getEnveloped({
-              ...ctxForReq.get(req.raw),
-              request: req.raw,
-              params,
-            })
-
-            const document = enveloped.parse(params.query)
-
-            const errors = enveloped.validate(enveloped.schema, document)
-
-            if (errors.length > 0) {
-              return { errors }
+    onRequest({ request, url, fetchAPI, endResponse }) {
+      const method = request.method.toLowerCase()
+      const token =
+        request.headers.get('X-GraphQL-Event-Stream-Token') ||
+        url.searchParams.get('token')
+      const acceptHeader = request.headers.get('Accept')
+      if (token != null) {
+        tokenByRequest.set(request, token)
+        if (acceptHeader?.includes('text/event-stream') && method === 'get') {
+          const encoder = new fetchAPI.TextEncoder()
+          endResponse(
+            new fetchAPI.Response(
+              map((str: string) => encoder.encode(str))(
+                pubsub.subscribe('graphql-sse-subscribe', token),
+              ) as unknown as BodyInit,
+              {
+                status: 200,
+                headers: {
+                  'Content-Type': 'text/event-stream',
+                },
+              },
+            ),
+          )
+        }
+      }
+      if (method === 'delete') {
+        const operationId = url.searchParams.get('operationId')
+        if (operationId) {
+          pubsub.publish('graphql-sse-unsubscribe', operationId, true)
+        }
+        endResponse(
+          new fetchAPI.Response(null, {
+            status: 204,
+          }),
+        )
+      }
+      if (method === 'put') {
+        const token = fetchAPI.crypto.randomUUID()
+        endResponse(
+          new fetchAPI.Response(token, {
+            status: 201,
+            statusText: 'Created',
+          }),
+        )
+      }
+    },
+    onParams({ request, params }) {
+      if (tokenByRequest.has(request) && params?.extensions?.operationId) {
+        operationIdByRequest.set(request, params.extensions.operationId)
+      }
+    },
+    onResultProcess({ request, result, fetchAPI, endResponse }) {
+      const token = tokenByRequest.get(request)
+      if (token) {
+        const operationId = operationIdByRequest.get(request)
+        // Batching is not supported by GraphQL SSE yet
+        if (operationId && !Array.isArray(result)) {
+          Promise.resolve().then(async () => {
```

Comment:
Why `Promise.resolve().then`?

Reply:
Not to block this response;
https://github.com/dotansimha/graphql-yoga/pull/2458/files#diff-15a5a73f50591db6e847522e52dfcc968134fc887b83d005594c430a3629c94dR128